### ISSUE TITLE
Export bundles fails when too many charms (limit unknown)

### ISSUE THREAD

**Issue by SaMnCo** [juju/juju-gui#1464]
Hi All, 

Using the demo GUI, I can't export large bundles. I have errors in the GUI console when I hit export like: 

```
WebSocket connection to 'wss://demo.jujucharms.com/ws' failed: Invalid frame header
Failed to load resource: the server responded with a status of 502 (Proxy Error)
XMLHttpRequest cannot load https://api.jujucharms.com/charmstore/v4/trusty/neutron-api-23/meta/any?inc…nifest&include=stats&include=charm-related&include=extra-info&include=tags. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://demo.jujucharms.com' is therefore not allowed access. The response had HTTP status code 502.
Object
WebSocket connection to 'wss://demo.jujucharms.com/ws' failed: Invalid frame header
Uncaught TypeError: Cannot read property 'get' of undefined
Object
Object
Object
Object
Uncaught TypeError: Cannot read property 'type' of undefined
Uncaught TypeError: Cannot read property 'type' of undefined
```

To reproduce:
* Add a base OpenStack model
* Add a Hadoop + rsyslog model and add relations to all OpenStack nodes 
* Add an ELK model, and make relations between logstash and all OpenStack services
* Add a Zabbix Model and add relation to all nodes. 
At some point, impossible to export. Unclear when. 

Thanks,

**Comment by urosj** [comment]
Interesting, @SaMnCo, we'll look into this. Do you think it's required to create all those relations for the export to fail?

**Comment by SaMnCo** [comment]
Not sure. Actually if you start with the attached
[bundle.txt](https://github.com/juju/juju-gui/files/165099/bundle.txt)

Then just add the OpenStack base, and try to export again. It will fail (at least it did on my Chrome)

**Comment by kadams54** [comment]
@SaMnCo Can you re-check? We recently released a new GUI version (2.1.1) that seemed to resolve the problem in our own investigations. I just re-tried with the attached bundle + OpenStack base against demo.jujucharms.com and was able to export without error. Thanks!

**Comment by hatched** [comment]
Closing this as resolved, if you're able to reproduce the original issue please feel free to re-open. Thank you for submitting this issue.
